### PR TITLE
Screen class

### PR DESCRIPTION
**Pull request by houghj16**
## Summary:
- This pull request adds a check for the screen recorder cache path before starting the recording. If the cache path is empty or invalid, an exception will be thrown and the recording will not start. This is to prevent the recording from failing silently or corrupting the video file.

### Files changed:
- ScreenRecorder.cs: Added a method ValidateCachePath that checks if the cache path is empty or invalid and throws an exception if so. Called this method in the StartRecording method before creating the video file.
- ScreenRecorderTest.cs: Added a unit test TestValidateCachePath that verifies the behavior of the ValidateCachePath method.

### Changes:
- Added a new exception class ScreenRecorderException that inherits from Exception and has a custom message property.
- Added a new method ValidateCachePath in the ScreenRecorder class that checks if the cache path is empty or invalid and throws a ScreenRecorderException if so.
- Added a call to ValidateCachePath in the StartRecording method before creating the video file.
- Added a new unit test TestValidateCachePath in the ScreenRecorderTest class that verifies the behavior of the ValidateCachePath method.

### Impact:
- This change will improve the user experience by preventing the recording from failing silently or corrupting the video file if the cache path is empty or invalid.
- This change will also make the code more robust and maintainable by handling the edge case of an empty or invalid cache path.

### Test plan:
- [ ] Run the unit test TestValidateCachePath and verify that it passes.
- [ ] Run the screen recorder with a valid cache path and verify that it works as expected.
- [ ] Run the screen recorder with an empty or invalid cache path and verify that it throws an exception and does not start the recording.

### CODE REVIEW

File: ShareX.ScreenCaptureLib/ScreenRecording/FFmpegCaptureDevice.cs [hunk 1]
```diff
@@ -29,7 +29,6 @@ public class FFmpegCaptureDevice
     {
         public string Value { get; set; }
         public string Title { get; set; }
-        // New updates in capture
```
Comment (houghj16) [inline ShareX.ScreenCaptureLib/ScreenRecording/FFmpegCaptureDevice.cs]: New comment

File: ShareX.ScreenCaptureLib/ScreenRecording/ScreenGrab.cs [hunk 2]
```diff
@@ -28,8 +28,10 @@
 using System;
 using System.Diagnostics;
 using System.Drawing;
+using System.IO;
```
Comment (houghj16) [inline ShareX.ScreenCaptureLib/ScreenRecording/ScreenGrab.cs:31]: New comment
